src/fermat.jl

```diff
@@ -0,0 +1,39 @@
+
+using NumberTheoreticTransforms
+
+export fermat
+
+"""
+    fermat(convolved, kernel)
+
+Calculates deconvolution with Number Theoretic Transform.
+"""
+function fermat(convolved::AbstractArray{T, 1}, h::AbstractArray{T, 1}, g::T, q::T) where {T<:Integer}
+    N = length(convolved)
+    bias = div(q-1, 2) + 1 #negative numbers are represeted by the upper half of [0, q) range
+    H = fnt(h, g, q)
+    H_inv = invmod.(H, q)
+    D = fft(h) |> prod |> real |> T
+    Dm = mod(D, q)
+    xm = ifnt(H_inv .* fnt(convolved, g, q), g, q)
+    x0 = mod.(Dm * xm, q)
+    x0[x0 .>= bias] .-= q
+    x = x0
```

Comment:
`x` will be exactly the same thing as `x0`, this means that chaning `x0` will have the same effect on `x` and vice versa.  Is this what you want?

Reply:
Yes, this assignment is just to follow variable naming form the paper, x0 is not modified later, and new variable xj is introduced later int the loop.